statements.tex

```diff
@@ -11,79 +11,96 @@
 
 \begin{quote}
 \pnum
-For a range-based for statement of the form
+\oldoldtxt{For a}\newnewtxt{The} range-based \tcode{for} statement \oldoldtxt{of the form}
 
-\begin{bnf}
-\terminal{for} \terminal{(} for-range-declaration \terminal{:} expression \terminal{)} statement
-\end{bnf}
+\begin{ncbnf}
+\terminal{for (} for-range-declaration \terminal{:} \oldoldtxt{expression}\newnewtxt{for-range-initializer} \terminal{)} statement
+\end{ncbnf}
 %
-let \grammarterm{range-init} be equivalent to the expression surrounded by parentheses
+\begin{removedblock}
+\oldoldtxt{let \grammarterm{range-init} be equivalent to the expression surrounded by parentheses}
```

Comment:
Footnote 89 here ("this ensures that a top-level comma operator cannot be reinterpreted [...]") also needs to be shown as deleted.